cpp/src/Slice/Grammar.y

```diff
@@ -1189,30 +1032,24 @@ return_type
 ;
 
 // ----------------------------------------------------------------------
-operation_preamble
```

Comment:
We used to have duplicates of every rule, one normal, and one with `idempotent` in front.
I added an intermediate rule that matches an optional `idempotent` keyword to eliminate this duplication.